[NO GBP]Fixes ethereal charging and recharge station charge speed.

Pickle-Coding opened the pull request:
## About The Pull Request
Fixes many instances of things not charging ethereals properly. Scales all things that are meant for charging/taking from the ethereal stomach by STANDARD_CELL_CHARGE, so we never run into this issue again. Ethereal stomachs now store a cell inside them, and uses that for the charge instead of tracking a variable. Fixes recharging stations not being able to charge ethereal stomachs. The ethereal signal proc attempted to feed a callback datum to adjust_charge(), which caused a runtime. Changes that by invoking the charge_cell callback instead.

Also fixes recharge station charging speed. They weren't converted correctly. Also formats their charging speed in their description, and displays power rather than referencing cycles.
## Why It's Good For The Game
So ethereals charge properly.

Closes #82470
## Changelog
:cl:
fix: Fixes many instances of energy sources for ethereals supplying a thousand times less energy than intended.
fix: Fixes recharging stations not being able to charge ethereals.
fix: Fixes recharge stations charging too fast.
qol: Recharge stations display their recharging speed in formatted power, rather than unformatted energy per cycle.
/:cl:

Pickle-Coding committed:
Fixes ethereal charging related stuff.

Creates a new cell /obj/item/stock_parts/cell/ethereal. Used for ethereal stomachs.

Ethereal stomachs now use a cell to track charge, rather than its own variable.

Converts many things that charge/discharge an ethereal to the joule. Also makes them and every other ethereal charge related thing scale with STANDARD_CELL_CHARGE.

Ethereal stomach overcharge tesla zapping is now based on how much energy gets discharged from the cell. Makes the proportion of the cell discharge to use for zapping inversely scale with the STANDARD_CELL_CHARGE define, so the energy of the zap stays the same when STANDARD_CELL_CHARGE changes.

Fixes recharger not charging ethereals properly, by adding the callback parameter for the ethereal signal proc and and invoking it with the ethereal stomach's cell as the cell parameter, instead of attempting to call adjust_charge with the callback as a parameter.

Also fixes ethereals discharging twice the amount of energy when charging an APC.

MrMelbert commented on code/modules/surgery/organs/internal/stomach/stomach_ethereal.dm:
```diff
@@ -3,11 +3,19 @@
 
+/obj/item/organ/internal/stomach/ethereal/Initialize(mapload)
+	. = ..()
+	cell = new/obj/item/stock_parts/cell/ethereal(src)
```
Do we want this in the contents of the stomach?

It means it'll be included in `get_all_contents` checks which in means stuff like "Electrical Activity" from botany would affect it. (And MAYBE EMPs?) Might be safer to have it be `nullspace`d

Pickle-Coding replied:
Do I just put null in the parameter?

Pickle-Coding committed:
Also fix recharge station charging speed.

The conversion for the recharge station wasn't done correctly. Converts it correctly and simplifies the math. Also makes the recharge station display the formatted charging speed.

Pickle-Coding committed:
Nullspace ethereal stomach cells.

Makes ethereal stomach cells instantiate in nullspace.

san7890 commented on code/modules/surgery/organs/internal/stomach/stomach_ethereal.dm:
```diff
@@ -3,11 +3,19 @@
+
+/obj/item/organ/internal/stomach/ethereal/Destroy()
+	QDEL_NULL(cell)
+	. = ..()
```
```suggestion
	return ..()
```

san7890 commented on code/modules/surgery/organs/internal/stomach/stomach_ethereal.dm:
```diff
@@ -3,11 +3,19 @@
 
+/obj/item/organ/internal/stomach/ethereal/Initialize(mapload)
+	. = ..()
+	cell = new/obj/item/stock_parts/cell/ethereal(null)
```
```suggestion
	cell = new /obj/item/stock_parts/cell/ethereal(null)
```

san7890 reviewed (changes requested):
style nits

Pickle-Coding committed:
Update stomach_ethereal.dm

Co-authored-by: san7890 <the@san7890.com>

Pickle-Coding committed:
Update stomach_ethereal.dm

Co-authored-by: san7890 <the@san7890.com>
